### PR TITLE
Allow user to specify promotion namespace when do fast-forward

### PR DESCRIPTION
**Pull request by skeeey**
fix the issue https://github.com/openshift/ci-tools/issues/1118, We want to allow user to specify their own promotion namespace when they do fast-forward

### CODE REVIEW

File: pkg/promotion/promotion.go [hunk 1]
```diff
@@ -121,11 +123,19 @@ func (o *Options) Validate() error {
...
 
 func (o *Options) matches(configuration *cioperatorapi.ReleaseBuildConfiguration) bool {
-	return PromotesOfficialImages(configuration) && configuration.PromotionConfiguration.Name == o.CurrentRelease
+	if isDisabled(configuration) {
+		return false
+	}
+
+	promotionNamespace := extractPromotionNamespace(configuration)
+	promotionName := extractPromotionName(configuration)
+	return (promotionNamespace == okdPromotionNamespace && promotionName == okd40Imagestream) ||
```
[the remark this answers is not in the document]
Comment (skeeey) [reply pkg/promotion/promotion.go:137]: if using `RefersToOfficialImage` function, the expression will be 

```go
return RefersToOfficialImage(promotionName, promotionNamespace) ||
		promotionNamespace == o.CurrentPromotionNamespace &&
			promotionName == o.CurrentRelease
```

I think this is a bit complex, but, yes, if I use `RefersToOfficialImage`, I can keep the determining official image in one place, so change it

[the remark this answers is not in the document]
Comment (skeeey) [reply pkg/promotion/promotion.go]: @alvaroaleman any more comments?

File: pkg/promotion/promotion.go [hunk 2]
```diff
@@ -121,11 +123,20 @@ func (o *Options) Validate() error {
 
 func (o *Options) Bind(fs *flag.FlagSet) {
 	fs.StringVar(&o.CurrentRelease, "current-release", "", "Configurations targeting this release will get branched.")
+	fs.StringVar(&o.CurrentPromotionNamespace, "current-promotion-namespace", ocpPromotionNamespace, "The promotion namespace of the current release.")
 	o.ConfirmableOptions.Bind(fs)
 }
 
 func (o *Options) matches(configuration *cioperatorapi.ReleaseBuildConfiguration) bool {
```
Comment (stevekuznetsov) [inline pkg/promotion/promotion.go:130]: Could we write something like:

```go
var imagesMatch bool
if o.CurrentPromotionNamespace == "" {
    imagesMatch = PromotesOfficialImages(configuration)
} else {
    imagesMatch = PromotesImagesInto(configuration, o.CurrentPromotionNamespace)
}

return imagesMatch && configuration.PromotionConfiguration.Name == o.CurrentRelease
```

Where we can factor `PromotesOfficialImages` to use `PromotesImagesInto` with `"ocp"` constant and keep the business logic for this in promotion.go

Comment (skeeey) [reply pkg/promotion/promotion.go:138]: yes, I can introduce the `PromotesImagesInto`  function and this makes the logic more clear

**Comment by skeeey** [comment]
@stevekuznetsov @alvaroaleman any more comments? if no comments, would you help me merge this pr, because our team has a task to dependent on this function, and we want to finish it in this week :)

**Pull request merged**